File: 02_ProgrammingFundamentals/_05_Lists/_02_Exercise/_04_ListOperations.java
package _02_ProgrammingFundamentals._05_Lists._02_Exercise;

import java.util.Arrays;
import java.util.List;
import java.util.Scanner;
import java.util.stream.Collectors;

public class _04_ListOperations {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        // Read input values
        List<Integer> numbersList = Arrays.stream(scan.nextLine().split("\\s+")).map(Integer::parseInt).collect(Collectors.toList());
        String console = scan.nextLine();

        // Transform the list based on the given commands
        while (!console.equals("End")) {
            String[] commandLine = console.split("\\s+");

            if (commandLine[0].equals("Add")) {
                numbersList.add(Integer.parseInt(commandLine[1]));
            } else if (commandLine[0].equals("Insert")) {
                if (Integer.parseInt(commandLine[2]) >= 0 && Integer.parseInt(commandLine[2]) < numbersList.size()) {
                    numbersList.add(Integer.parseInt(commandLine[2]), Integer.parseInt(commandLine[1]));
                } else {
                    System.out.println("Invalid index");
                }
            } else if (commandLine[0].equals("Remove")) {
                if (Integer.parseInt(commandLine[1]) >= 0 && Integer.parseInt(commandLine[1]) < numbersList.size()) {
                    numbersList.remove(Integer.parseInt(commandLine[1]));
                } else {
                    System.out.println("Invalid index");
                }
            } else if (commandLine[0].equals("Shift")) {
                int n = Integer.parseInt(commandLine[2]);

                if (commandLine[1].equals("left")) {
                    for (int i = 0; i < n; i++) {
                        numbersList.add(numbersList.get(0));
                        numbersList.remove(0);
                    }
                } else if (commandLine[1].equals("right")) {
                    for (int i = 0; i < n; i++) {
                        numbersList.add(0,numbersList.get(numbersList.size() - 1));
                        numbersList.remove(numbersList.size() - 1);
                    }
                }
            }

            console = scan.nextLine();
        }

        //Print result
        numbersList.stream().forEach(e -> System.out.print(e + " "));

        scan.close();
    }
}

File: 24_C#Fundamentals/_05_Lists/_02_Exercise/_04_ListOperations/_04_ListOperations.cs
using System;
using System.Xml.Linq;

public class _04_ListOperations
{
    public static void Main(string[] args)
    {
        // Read input values
        List<int> numbersList = Console.ReadLine().Split().Select(int.Parse).ToList();
        string console = Console.ReadLine();

        // Transform the list based on the given commands
        while (!console.Equals("End"))
        {
            string[] commandLine = console.Split();

            if (commandLine[0].Equals("Add"))
            {
                numbersList.Add(int.Parse(commandLine[1]));
            }
            else if (commandLine[0].Equals("Insert"))
            {
                if (int.Parse(commandLine[2]) >= 0 && int.Parse(commandLine[2]) < numbersList.Count)
                {
                    numbersList.Insert(int.Parse(commandLine[2]), int.Parse(commandLine[1]));
                }
                else
                {
                    Console.WriteLine("Invalid index");
                }
            }
            else if (commandLine[0].Equals("Remove"))
            {
                if (int.Parse(commandLine[1]) >= 0 && int.Parse(commandLine[1]) < numbersList.Count)
                {
                    numbersList.RemoveAt(int.Parse(commandLine[1]));
                }
                else
                {
                    Console.WriteLine("Invalid index");
                }
            }
            else if (commandLine[0].Equals("Shift"))
            {
                int n = int.Parse(commandLine[2]);

                if (commandLine[1].Equals("left"))
                {
                    for (int i = 0; i < n; i++)
                    {
                        numbersList.Add(numbersList[0]);
                        numbersList.RemoveAt(0);
                    }
                }
                else if (commandLine[1].Equals("right"))
                {
                    for (int i = 0; i < n; i++)
                    {
                        numbersList.Insert(0, numbersList[numbersList.Count - 1]);
                        numbersList.RemoveAt(numbersList.Count - 1);
                    }
                }
            }

            console = Console.ReadLine();
        }

        // Print result
        Console.WriteLine(string.Join(" ", numbersList));
    }
}
